enclave_build/src/cache.rs

```diff
@@ -0,0 +1,784 @@
+        // the cache by checking the digest
+        for layer_obj in layers_vec {
+            // Read the layer digest from the manifest
+            let layer_digest: String = layer_obj
+                .get("digest")
+                .ok_or_else(|| {
+                    Error::ImageNotCached("Image layer digest not found in manifest".to_string())
+                })?
+                .as_str()
+                .ok_or_else(|| Error::ImageNotCached("Layer info extract error".to_string()))?
+                .strip_prefix("sha256:")
+                .ok_or_else(|| Error::ImageNotCached("Layer info extract error".to_string()))?
+                .to_string();
+
+            // Get the cached layer file matching the digest
+            // If not present, then a layer file is missing, so return Error
+            let mut layer_file = cached_layers
+                .get(&layer_digest)
+                .ok_or_else(|| Error::ImageNotCached("Layer missing from cache.".to_string()))?;
+            let mut layer_bytes = Vec::new();
+            layer_file
+                .read_to_end(&mut layer_bytes)
+                .map_err(|_| Error::ImageNotCached("Failed to read layer".to_string()))?;
+
+            let calc_digest = format!("{:x}", sha2::Sha256::digest(layer_bytes.as_slice()));
+
+            // Check that the digests match
+            if calc_digest != layer_digest {
+                return Err(Error::ImageNotCached("Layer not valid".to_string()));
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Returns the default root folder path of the cache.
+    ///
+    /// The default cache path is {XDG_DATA_HOME}/.nitro_cli/container_cache, and if the env
+    /// variable is not set, {HOME}/.local/share/.nitro_cli/container_cache is used.
+    pub fn get_default_cache_root_path() -> Result<PathBuf> {
+        // Try to use XDG_DATA_HOME as default root
+        let root = match std::env::var_os(CACHE_ROOT_FOLDER) {
+            Some(val) => val
+                .into_string()
+                .map_err(|err| Error::CacheRootPath(format!("{:?}", err)))?,
+            // If XDG_DATA_HOME is not set, use {HOME}/.local/share as specified in
+            // https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+            None => {
+                let home_folder = std::env::var_os("HOME").ok_or_else(|| {
+                    Error::CacheRootPath("HOME environment variable is not set.".to_string())
+                })?;
+                format!(
+                    "{}/.local/share/",
+                    home_folder
+                        .into_string()
+                        .map_err(|err| Error::CacheRootPath(format!("{:?}", err)))?
+                )
+            }
+        };
+
+        let mut path = PathBuf::from(root);
+        // Add the additional path to the root
+        path.push(".nitro_cli/container_cache");
```

Comment:
use `aws-nitro-enclaves-cli` instead of `.nitro_cli`. There's already another `nitro-cli` package in Ubuntu/Fedora and they shouldn't collide.